lib/dor/workflow/client/lifecycle_routes.rb

```
@@ -10,33 +11,101 @@ def initialize(requestor:)
         # @return [Time] when the milestone was achieved.  Returns nil if the milestone does not exis
         #
-        def active_lifecycle(repo, druid, milestone_name, version: nil)
-          lifecycle(repo, druid, milestone_name, version: version, active_only: true)
+        def active_lifecycle(*args)
+          case args.size
+          when 4
+            Deprecation.warn(self, 'you provided 4 args, but active_lifecycle now takes kwargs')
+            (repo, druid, milestone_name) = args[0..2]
+            version = args[3][:version]
+          when 3
+            Deprecation.warn(self, 'you provided 3 args, but active_lifecycle now takes kwargs')
+            (repo, druid, milestone_name) = args
+            version = nil
+          when 2
+            Deprecation.warn(self, 'you provided 2 args, but active_lifecycle now takes kwargs')
+            (druid, milestone_name) = args
+            version = nil
+          when 1
+            opts = args.first
+            repo = opts[:repo]
+            druid = opts[:druid]
+            milestone_name = opts[:milestone_name]
+            version = opts[:version]
+          else
+            raise ArgumentError, 'wrong number of arguments, must be 1-4'
+          end
+
+          Deprecation.warn(self, 'passing the repo parameter to active_lifecycle is no longer necessary. This will raise an error in dor-workflow-client version 4') if repo
+
+          lifecycle(druid: druid, milestone_name: milestone_name, version: version, active_only: true)
         end
 
         # @return [Array<Hash>]
-        def milestones(repo, druid)
-          doc = query_lifecycle(repo, druid, active_only: false)
+        def milestones(*args)
+          case args.size
+          when 2
+            Deprecation.warn(self, 'you provided 2 args, but active_lifecycle now takes kwargs')
+            (repo, druid) = args
+          when 1
+            opts = args.first
+            repo = opts[:repo]
+            druid = opts[:druid]
```

This should probably be `opts.fetch(:druid)` because it should raise an error if you don't pass it.

---

Done.